src/main/java/org/candlepin/resource/ProductResource.java

```diff
@@ -142,15 +143,14 @@ public ProductCertificateDTO getProductCertificate(String productUuid) {
             return null;
         }
 
-        ResultIterator<Owner> iterator = this.ownerCurator.getOwnersWithProducts(productUuids).iterate();
+        Iterator<Owner> iterator = this.ownerCurator.getOwnersWithProducts(productUuids).iterator();
```

Comment:
Sorry for noticing this late, but we can turn all of this into a stream with a couple mappers and/or functions:

```
        Function<Owner, JobConfig> jobConfigMapper = (owner) -> {
            return RefreshPoolsJob.createJobConfig()
                .setOwner(owner)
                .setLazyRegeneration(lazyRegen);
        };

        Function<JobConfig, AsyncJobStatus> jobQueueMapper = (config) -> {
            try {
                return this.jobManager.queueJob(config);
            }
            catch (Exception e) {
                log.debug("Exception occurred while queueing job: {}", config, e);

                return new AsyncJobStatus()
                    .setName(RefreshPoolsJob.JOB_NAME)
                    .setState(AsyncJobStatus.JobState.FAILED)
                    .setJobResult(e.toString());
            }
        }

        return this.ownerCurator.getOwnersWithProducts(productUuids)
            .stream()
            .map(jobConfigMapper)
            .map(jobQueueMapper)
            .map(this.translator.getStreamMapper(AsyncJobStatus.class, AsyncJobStatusDTO.class));
```

Reply:
Done.